File: calculator-csharp/CalculatorLib/CalculatorService/Calculator.cs
using System;
using System.Linq;


namespace CalculatorService
{
    /// <summary>Main class to calculate arithmetic expression</summary>
    public static class Calculator
    {
        public static Number Calc(string text)
        {
            var lexer = new Lexer(text);
            var tokens = lexer.GenerateTokens();

            var parser = new Parser(tokens.ToList());
            var tree = parser.Parse();

            var interpreter = new Interpreter();

            return tree.accept(interpreter);
        }
    }
}

File: calculator-java/library/src/main/java/com/example/calculator/service/Calculator.java
package com.example.calculator.service;

public class Calculator {

    public static Number calc(String text) {
        var lexer = new Lexer(text);
        var tokens = lexer.generateTokens();

        var parser = new Parser(tokens);
        var tree = parser.parse();

        var interpreter = new Interpreter();

        return tree.accept(interpreter);
    }

}
